catalog/ui/search-ui/standard/src/main/webapp/js/widgets/cesium.mapclustering.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/* global define */
+define(['marionette',
+        'cesium',
+        'jquery'
+], function(Marionette, Cesium, $) {
+
+    var DEFAULT_PIXEL_DISTANCE = 100;
+
+    var MAX_RADIUS_DISTANCE = 2500000;
+
+    var MIN_CAMERA_HEIGHT = 5;
+
+    var MARKER_COLOR = Cesium.Color.RED;
+
+    var MapClustering = {};
+
+    MapClustering = Marionette.Controller.extend({
+        initialize: function() {
+            this.clusters = [];
+        },
+        setResultLists : function(entityList) {
+            this.entities = entityList;
+        },
+        setViewer : function(mapView) {
+            this.viewer = mapView;
+        },
+        clusteringAlgorithm : function (cameraHeight) {
+            if(typeof this.entities === "undefined") {
+                return;
+            } else if(cameraHeight < MIN_CAMERA_HEIGHT) {
+                this.resetBillboard();
+            } else {
+                this.removeClusters();
+                var models = this.entities.geoController.mapViews.collection;
+                var views = this.entities.geoController.mapViews.children;
+                var modelsLength = models.length;
+
+                for(var c = 0; c < modelsLength; c++) {
```

Comment:
Backbone collections have a forEach you can use.

http://backbonejs.org/#Collection-Underscore-Methods